api/tests/routes/backoffice_v3/offerers_test.py

```diff
@@ -383,3 +387,670 @@ def send_comment_offerer_request(self, client, legit_user, offerer, comment):
+            assert len(rows) == 1
+            assert rows[0]["ID"] == str(user_offerer.offerer.id)
+            assert rows[0]["Nom de la structure"] == user_offerer.offerer.name.upper()
+            assert rows[0]["État"] == "Nouvelle"
+            assert rows[0]["Date de la demande"] == "28/10/2022"
+            assert rows[0]["Dernier commentaire"] == ""
+
+        @override_features(WIP_ENABLE_BACKOFFICE_V3=True)
+        def test_payload_content_no_action(self, authenticated_client):
+            # given
+            user_offerer = offerers_factories.UserNotValidatedOffererFactory(
+                offerer__dateCreated=datetime.datetime(2022, 10, 3, 11, 59),
+            )
+
+            # when
+            with assert_no_duplicated_queries():
+                response = authenticated_client.get(
+                    url_for("backoffice_v3_web.validate_offerer.list_offerers_to_validate")
+                )
+
+            # then
+            assert response.status_code == 200
+            rows = html_parser.extract_table_rows(response.data)
+            assert len(rows) == 1
+            assert rows[0]["ID"] == str(user_offerer.offerer.id)
+            assert rows[0]["Nom de la structure"] == user_offerer.offerer.name.upper()
+            assert rows[0]["État"] == "Nouvelle"
+            assert rows[0]["Date de la demande"] == "03/10/2022"
+            assert rows[0]["Dernier commentaire"] == ""
+
+        @override_features(WIP_ENABLE_BACKOFFICE_V3=True)
+        @pytest.mark.parametrize(
+            "total_items, pagination_config, expected_total_pages, expected_page, expected_items",
+            (
+                (31, {"per_page": 10}, 4, 1, 10),
+                (31, {"per_page": 10, "page": 1}, 4, 1, 10),
+                (31, {"per_page": 10, "page": 3}, 4, 3, 10),
+                (31, {"per_page": 10, "page": 4}, 4, 4, 1),
+                (20, {"per_page": 10, "page": 1}, 2, 1, 10),
+                (10, {"page": 1}, 1, 1, 10),
+                (10, {"per_page": 25, "page": 1}, 1, 1, 10),
+            ),
+        )
+        def test_list_pagination(
+            self,
+            authenticated_client,
+            total_items,
+            pagination_config,
+            expected_total_pages,
+            expected_page,
+            expected_items,
+        ):
+            # given
+            for _ in range(total_items):
+                offerers_factories.UserNotValidatedOffererFactory()
+
+            # when
+            response = authenticated_client.get(
+                url_for("backoffice_v3_web.validate_offerer.list_offerers_to_validate", **pagination_config)
+            )
+
+            # then
+            assert response.status_code == 200
+            assert html_parser.count_table_rows(response.data) == expected_items
+            assert html_parser.extract_pagination_info(response.data) == (
+                expected_page,
+                expected_total_pages,
+                total_items,
+            )
```

Comment:
Est-ce qu'on a vraiment besoin de tester la pagination ici ?
A la place, on pourrait tester la fonction `pagination_links` directement. J'ai peur qu'on ne s'en sorte pas si on commence à tester la pagination de cette manière, sur chaque page qui l'utilise =/

Reply:
ouais j'avais eu la même remarque par Damien à l'époque et j'avais noté cette suggestion dans un ticket d'amélioration des tests qui doit toujours être dans le backlog.